diff --git a/wiringPi/drcNet.c b/wiringPi/drcNet.c
index 0fc5d2b..0e0b473 100644
--- a/wiringPi/drcNet.c
+++ b/wiringPi/drcNet.c
@@ -31,7 +31,6 @@
 #include <netdb.h>
 #include <string.h>
 #include <errno.h>
-#include <crypt.h>


 #include "wiringPi.h"
diff --git a/wiringPi/wiringPi.c b/wiringPi/wiringPi.c
index 5c6a0f6..d4a4e96 100644
--- a/wiringPi/wiringPi.c
+++ b/wiringPi/wiringPi.c
@@ -1777,11 +1777,11 @@ int digitalRead (int pin)
       return LOW ;
 
     if (PI_MODEL_5 == RaspberryPiModel) {
-      switch(gpio[2*pin] & RP1_STATUS_LEVEL_MASK) {
-        default: // 11 or 00 not allowed, give LOW!
-        case RP1_STATUS_LEVEL_LOW:  return LOW ;
-        case RP1_STATUS_LEVEL_HIGH: return HIGH ;
-      }
+//      switch(gpio[2*pin] & RP1_STATUS_LEVEL_MASK) {
+//        default: // 11 or 00 not allowed, give LOW!
+//        case RP1_STATUS_LEVEL_LOW:  return LOW ;
+//        case RP1_STATUS_LEVEL_HIGH: return HIGH ;
+//      }
     } else {
       if ((*(gpio + gpioToGPLEV [pin]) & (1 << (pin & 31))) != 0)
         return HIGH ;
